clang/tools/driver/CacheLauncherMode.cpp

```diff
@@ -30,15 +32,50 @@ static bool isSameProgram(StringRef clangCachePath, StringRef compilerPath) {
   return CacheStat.getUniqueID() == CompilerStat.getUniqueID();
 }
 
+static bool shouldCacheInvocation(ArrayRef<const char *> Args,
+                                  IntrusiveRefCntPtr<DiagnosticsEngine> Diags) {
+  SmallVector<const char *, 128> CheckArgs(Args.begin(), Args.end());
+  // Make sure "-###" is not present otherwise we won't get an object back.
```

Comment:
> There are other options I think if you go this way. Like `-ccc-print-phases`

I needed to handle `-###` because I use it for testing, it's not clear to me whether handling `-ccc-print-phases` is important or not.
 
> I wonder what is the overhead for creating a compiler invocation?

Seems to be 2~3 ms on M1Pro.

> Alternatively, you can just create a clang OptTable and parse arg to decide. The downside is it needs to stay in sync with the clang frontend.

There is logic in the driver to determine whether modules are actually used or not, so I wanted to take advantage of that instead of just checking for the presence of flags.